test/is_palindrome_test.cpp

```diff
@@ -18,6 +18,7 @@
 #include <iostream>
 #include <list>
 #include <vector>
+#include <iterator>
```

Comment:
`std::begin/end` is came from C++11, so I think those should be replaced with `boost::begin/end`.